openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -1158,7 +1162,7 @@ impl UhProcessor<'_, TdxBacked> {
             .with_vector(2)
             .with_interruption_type(INTERRUPT_TYPE_NMI);
 
-        self.backing.cvm.lapics[GuestVtl::Vtl0].halted = false;
+        self.backing.cvm.lapics[GuestVtl::Vtl0].activity = MpState::Running;
```

Comment:
Couldn't we be in the WaitForSipi state here?

Reply:
Do we expect that an NMI would come to a processor that's still in WaitForSipi? Would such an event not kick it into Running? I guess that's an x86 spec question?

Reply:
ChatGPT says that the expected behavior is we should ignore NMIs and interrupts while in WaitForSipi, I'll add that.